routing_algorithms/many_to_many.hpp

```diff
@@ -67,11 +67,12 @@ class ManyToManyRouting final
     ~ManyToManyRouting() {}
 
     std::shared_ptr<std::vector<EdgeWeight>>
-    operator()(const PhantomNodeArray &phantom_nodes_array) const
+    operator()(const PhantomNodeArray &phantom_targets_array, const PhantomNodeArray &phantom_sources_array = PhantomNodeArray(0)) const
```

Comment:
Don't provide default values here. Parameter order should be `const PhantomNodeArray& sources, const PhantomNodeArray& targets`.